jetpack-start/wp-cli-api.php

```diff
@@ -0,0 +1,221 @@
+	const API_ERROR_USER_PERMISSIONS = 'User does not have permission to administer the given site.';
+
+	/**
+	 * Cancel a Jetpack Start subscription
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--jetpack_site_id=<site_id>]
+	 * : WP.com site ID for the shadow site. If not specified, grabs the site ID from Jetpack.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp jetpack-start api cancel
+	 *     wp jetpack-start api cancel --jetpack_site_id=999999999
+	 *
+	 */
+	public function cancel( $args, $assoc_args ) {
+		$site_id = WP_CLI\Utils\get_flag_value( $assoc_args, 'jetpack_site_id', false );
+
+		if ( empty( $site_id ) ) {
+			$site_id = Jetpack_Options::get_option( 'id' );
+			if ( empty( $site_id ) ) {
+				WP_CLI::error( 'Could not get WP.com blog_id from the local Jetpack install. Please specify the `--jetpack_site_id` of the Jetpack Shadow site to cancel the subscription. You can get this from the Jetpack Debugger or using WP.com Network Admin.' );
+			}
+		}
+
+		$data = $this->api_cancel_subscription( $site_id );
+		if ( is_wp_error( $data ) ) {
+			WP_CLI::error( $data->get_error_message() );
+		}
+
+		WP_CLI::success( sprintf( 'Cancelled subscription for site_id = %s (body: %s)', $site_id, var_export( $data, true ) ) );
+	}
+
+	/**
+	 * Connect the site using Jetpack Start
+	 *
+	 * Creates a machine user (if needed) and then uses the Jetpack Start API to initialize a connection and sets up the necessary local bits.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--force]
+	 * : Provision even if Jetpack is already connected.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp jetpack-start api connect
+	 *     wp jetpack-start api connect --force
+	 *
+	 */
+	public function connect( $args, $assoc_args ) {
+		if ( ! defined( 'WPCOM_VIP_JP_START_API_BASE' ) || ! defined( 'WPCOM_VIP_JP_START_API_TOKEN' ) ) {
+			WP_CLI::error( 'Jetpack Start API constants (`WPCOM_VIP_JP_START_API_BASE` and/or `WPCOM_VIP_JP_START_API_TOKEN`) are not defined. Please check the `vip-secrets` file to confirm they\'re there, accessible, and valid.' );
+		}
+
+		if ( defined( 'JETPACK_DEV_DEBUG' ) && true === JETPACK_DEV_DEBUG ) {
+			WP_CLI::warning( 'JETPACK_DEV_DEBUG mode is enabled. Please remove the constant after connection.' );
+		}
+
+		$force_connection = WP_CLI\Utils\get_flag_value( $assoc_args, 'force', false );
+		if ( ! $force_connection && Jetpack::is_active() ) {
+			WP_CLI::error( 'Jetpack is already active; bailing. Run this command with `--force` to bypass this check or disconnect Jetpack before continuing.' );
+		}
+
+		WP_CLI::line( '-- Verifying VIP machine user exists (or creating one, if not)' );
+		$user = $this->maybe_create_user();
+		if ( is_wp_error( $user ) ) {
+			WP_CLI::error( $user->get_error_message() );
+		}
+
+		WP_CLI::line( '-- Fetching keys from Jetpack Start API' );
+		$data = $this->api_fetch_keys( $user );
+		if ( is_wp_error( $data ) ) {
+			$message = $data->get_error_message();
+			if ( self::API_ERROR_EXISTING_SUBSCRIPTION === $message ) {
```

Comment:
Unfortunately not. The error code returned from the API are pretty generic. I'll see if we can update them on the API side.